### ISSUE TITLE
pseudo-inverse matrix - Moore-Penrose

### ISSUE THREAD

**Issue by gomaja** [gonum/gonum#1842]
Hello, 
I was looking at issue [mat pseudo-inverse #409], and I found that it is useful to mention and see if such function can be developed.
In fact if the matrix is not invertible under the well known inversion of a square matrix, due to the fact that it is not square matrix (i.e. m!=n), it can have a pseudo inverse matrix under Moore-Penrose technique.
To demonstrate, an example of such function can be found elsewhere for example MATLAB https://www.mathworks.com/help/matlab/ref/pinv.html

Do the existing functions in gonum have such special inversion ?

**Comment by kortschak** [comment]
No, we do not provide this for the reasons that are given in #409. In the vast majority of cases inverses and pseudoinverses are used where a solve would be better.

If you have a specific case where an inverse is required not as part of a solve, please describe the use-case in a feature request.

**Comment by MarwanJADID** [comment]
In fact, the solve can solve the problem **iteratively**, and requires the output function values (error vector values).

The pseudo-inverse is an iterative free solution (but not necessarily the optimum), and is a generalized solution where no pre-information is required to for the output function (error vector values are not considered, only matrix inversion).

**Comment by kortschak** [comment]
Please provide a use case that you are trying to address that you cannot achieve with what already exists.

**Comment by MarwanJADID** [comment]
lets consider the system y=Ax such that the system is overdetermined linear system.
Given A, for a certain given y=y1, the system can be iteratively optimized to calculate x=x1. For another given y=y2, the iterative system calculates another x=x2.  
The pseudo-inverse calculates pinv(A) without the need to know y, and once done, any x can be calculated based on a given y with a simple matrix multiplication x=pinv(A)y without the need to iteratively optimize the system each time.

You may say that we can iteratively optimize the system for once, and in a certain way (not sure how to work it around) we can deduce the what we call pseudo inverse of A. And in fact this is not very true, since when using solve, the system is optimal for a certain objective function y, and this perfect optimality creates sensitivity (over-fitted). However the Moore-Penrose pseudo-inverse keeps the system solution non-overfitted and generalized for any objective (definitely within something that relates the quantities of the objective function y in a common sense).

To summarize, the idea is to compute the overdetermined system relation pinv(A) only once, and deploy it for multiplication purpose only which reduce execution time and requires less resources.

**Comment by kortschak** [comment]
You can use the SVD which we provide to solve for the pseudo inverse. I don't think we are going to add this.
